pkg/azuredisk/nodeserver_v2.go

```diff
@@ -573,9 +573,10 @@ func (d *DriverV2) recoverMount(diskURI string) {
 		detachErr := d.crdProvisioner.UnpublishVolume(recoveryCtx, diskURI, d.NodeID, nil)
 		if detachErr != nil {
 			klog.Errorf("failed to unpublishVolume volume (%s) from node (%s): %v", diskURI, d.NodeID, detachErr)
+			d.markDetachComplete(diskURI, false)
```

Comment:
Since we are marking a failed result here as well, the name should be something like `recordDetachResult`

Reply:
Yeah, just made a change to accommodate more detachment state and in process of doing that, I renamed the function as well :)